File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example2/BaseFileHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example2;

public abstract class BaseFileHandler : IFileHandler
{
    private IFileHandler? _nextHandler;

    public IFileHandler SetNext(IFileHandler handler)
    {
        _nextHandler = handler;
        return handler;
    }

    public virtual void Handle(FileUploadRequest request)
    {
        if (_nextHandler != null)
        {
            _nextHandler.Handle(request);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example2/src/BaseFileHandler.ts
import { FileUploadRequest, IFileHandler } from "./FileUploadRequest";

export abstract class BaseFileHandler implements IFileHandler {
  private _nextHandler: IFileHandler | null = null;

  setNext(handler: IFileHandler): IFileHandler {
    this._nextHandler = handler;
    return handler;
  }

  handle(request: FileUploadRequest): void {
    if (this._nextHandler) {
      this._nextHandler.handle(request);
    }
  }
}
